example/lib/mail_page.dart

```diff
@@ -0,0 +1,200 @@
+          isUnread: true,
+        ),
+        Mail(
+          sender: 'Github',
+          sub: 'Showcase View',
+          msg: 'New star on your showcase view.',
+          date: '21 May ',
+          isUnread: false,
+        ),
+        Mail(
+          sender: 'Simform',
+          sub: 'Credit card Plugin',
+          msg: 'Check out our credit card plugin',
+          date: '19 May',
+          isUnread: true,
+        ),
+      ],
+    );
+  }
+
+  void _getData() {
+    _addData();
+    Future.delayed(const Duration(milliseconds: 1500));
+    setState(() {
+      apiState = ApiState.success;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.white,
+      body: SafeArea(
+        bottom: false,
+        child: Column(
+          children: <Widget>[
+            const SizedBox(
+              height: 20,
+            ),
+            const Padding(padding: EdgeInsets.only(top: 8)),
+            TextButton(
+              onPressed: _getData,
+              child: const Text('Tap here to get Data'),
+            ),
+            Visibility(
+              visible: apiState != ApiState.success,
+              child: DropdownButton<String>(
+                value: dropdownValue,
+                icon: const Icon(Icons.arrow_drop_down),
+                elevation: 16,
+                style: const TextStyle(
+                  color: Colors.blue,
+                  fontWeight: FontWeight.bold,
+                ),
+                underline: Container(
+                  height: 2,
+                  color: Colors.redAccent,
+                ),
+                onChanged: (String? newValue) {
+                  _changePlaceHolderImage(newValue);
+                  setState(() {
+                    dropdownValue = newValue!;
+                  });
+                },
+                items: <String>[
+                  'Image',
+                  'SVG',
+                  'Flare',
+                  'Rive',
+                  'Rive HTTP',
+                  'GIF'
+                ].map<DropdownMenuItem<String>>((String value) {
+                  return DropdownMenuItem<String>(
+                    value: value,
+                    child: Text(value),
+                  );
+                }).toList(),
+              ),
+            ),
+            Expanded(
+              child: mails.isNotEmpty
+                  ? ListView.builder(
+                      physics: const BouncingScrollPhysics(),
+                      itemBuilder: (context, index) {
+                        return MailTile(mails[index % mails.length]);
+                      })
+                  : EmptyListPlaceholder(
+                      state: apiState,
+                      extraPadding: const EdgeInsets.only(bottom: 10),
+                      loadingTitle: 'Loading Title',
+                      loadingSubtitle: 'Loading subtitle',
+                      emptyTitle: 'No Data Found',
```

Comment:
instead of individual parameters, lets make a class of config that includes title, subtitle, URL. and update parameters of the main widget like empty config, Loading config. make sure to give default values to the configurations.